packages/devextreme/testing/testcafe/tests/scheduler/layout/templates/appointmentTemplate.ts

```diff
@@ -66,3 +76,274 @@ fixture.disablePageReloads`Layout:Templates:appointmentTemplate`
+      appointmentDates = Array.from(
+        { length: 12 },
+        (_, index) => {
+          const result = new Date(2024, 0, 1, 8);
+          result.setHours(result.getHours() + index);
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'week':
+    case 'timelineWeek': {
+      if (scrollMode === 'standard' && groupOrientation === 'horizontal') {
+        resourceCount = 2;
+      }
+
+      appointmentDates = Array.from(
+        { length: 12 * 7 },
+        (_, index) => {
+          const result = new Date(2023, 11, 31, 8);
+          result.setDate(result.getDate() + Math.floor(index / 12));
+          result.setHours(result.getHours() + (index % 12));
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'workWeek':
+    case 'timelineWorkWeek': {
+      if (scrollMode === 'standard' && groupOrientation === 'horizontal') {
+        resourceCount = 2;
+      }
+
+      appointmentDates = Array.from(
+        { length: 12 * 5 },
+        (_, index) => {
+          const result = new Date(2024, 0, 1, 8);
+          result.setDate(result.getDate() + Math.floor(index / 12));
+          result.setHours(result.getHours() + (index % 12));
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'month':
+    case 'timelineMonth': {
+      appointmentDates = Array.from(
+        { length: 31 },
+        (_, index) => {
+          const result = new Date(2024, 0, 1, 8);
+          result.setDate(result.getDate() + index);
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'day':
+    case 'timelineDay':
+    default: {
+      appointmentDates = Array.from(
+        { length: 12 },
+        (_, index) => {
+          const result = new Date(2024, 0, 2, 8);
+          result.setHours(result.getHours() + index);
+          return result;
+        },
+      );
+
+      break;
+    }
+  }
+
+  const datesToCheck = [
+    appointmentDates[0],
+    appointmentDates[Math.floor(appointmentDates.length / 3)],
+    appointmentDates[appointmentDates.length - 1],
+  ];
+
+  const groupsToCheck = [
+    { groupId: 0 },
+    { groupId: Math.floor(resourceCount / 3) },
+    { groupId: resourceCount - 1 },
+  ];
+
+  test(`targetedAppointmentData should be correct with groups (viewType="${viewType}", groupOrientation="${groupOrientation}", scrollMode="${scrollMode}", rtlEnabled="${rtlEnabled}") (T1205120)`, async (t) => {
+    const resourceDataSource = Array.from({ length: resourceCount }, (_, index) => ({
+      id: index,
+      text: `Resource ${index}`,
+    }));
+
+    const appointments = resourceDataSource.reduce<Appointment[]>((acc, resource) => acc.concat(
+      appointmentDates
+        .map<Appointment>((date) => ({
+        text: resource.text,
+        startDate: date,
+        endDate: new Date(date.getTime() + HOUR / 2),
+        groupId: resource.id,
+      })),
+    ), []);
+
+    await createWidget('dxScheduler', {
```

Comment:
Why do you initialize the widget in the test body?
We usually initialize widgets in the testcafe's ```before``` hook

If you have arguments that are better than ```before``` hook let's discuss it
If not, let's move the widget initialization to the ```before``` to make it consistent with other test cases :)